kontrol/handlers.go

```diff
@@ -82,66 +83,73 @@ func (k *Kontrol) HandleRegister(r *kite.Request) (interface{}, error) {
 	every := onceevery.New(UpdateInterval)
 
 	ping := make(chan struct{}, 1)
-	closed := false
+	closed := int32(0)
+
+	kiteCopy := remote.Kite
```

Comment:
remote.Kite.Username is mutated at [line 141](https://github.com/koding/kite/pull/155/files#diff-02312926bbc7db566f479305ba8bb9f4R141) and read in the `updaterFunc` and `heartbeatArgs` goroutines - race. The copying makes the goroutines use their own copy - to fix the race.